### PR TITLE
fix nil errors in handler

### PR DESCRIPTION
**Pull request by luispdm**
- `ReplaceBlockTag` does not return an additional `*JSONRPCRequest`, as it already changes the receiving parameter "req"
- no need to run `CheckBlockNumberFromResponseFromHashCalls` when `response.Result` is `nil` (`CheckBlockNumberFromResponseFromHashCalls` also checks if `response.Result` is `nil` as a first instruction)
- do not include `result` in RPCResponse when there's an error
- include `result: null` when the underlying L1 node returns it

### CODE REVIEW

File: internal/core/worker/evolution/worker.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (w *worker) Run(ctx context.Context) error {
 				slog.Error(err.Error())
 				break
 			}
-            if lastBlock < startingBlock {
+			if lastBlock < startingBlock {
```
Comment (zorjak) [inline internal/core/worker/evolution/worker.go:71]: can you remove these changes from this PR?

File: internal/core/worker/evolution/worker_test.go [hunk 2]
```diff
@@ -225,7 +225,7 @@ func TestScanEvoChainOnce(t *testing.T) {
 			client, scanner := getMocks(t)
 			state, tx2 := getMocksFromState(t)
 
-            evoWorker := NewWorker(&tt.c, client, scanner, state)
+			evoWorker := NewWorker(&tt.c, client, scanner, state)
```
Comment (zorjak) [inline internal/core/worker/evolution/worker_test.go:228]: similar as for worker

**Review by zorjak — CHANGES_REQUESTED** [review]

File: internal/core/worker/evolution/worker.go (hunk 1 above)
Comment (luispdm) [reply internal/core/worker/evolution/worker.go:71]: As agreed in private, this change will stay in the PR

File: internal/core/worker/evolution/worker_test.go (hunk 2 above)
Comment (luispdm) [reply internal/core/worker/evolution/worker_test.go:228]: As agreed in private, this change will stay in the PR

**Review by zorjak — APPROVED** [review]

**Pull request merged**